### ISSUE TITLE
cmd-click does not work

### ISSUE THREAD

**Issue by ajaybeniwal** [facebook/nuclide#92]
I have installed all the packages and when i do cmd-click it does not navigate to the correct function definitions, is there any other setting which needs to be turned on

**Comment by hrnn** [comment]
This is happening to me as well with `.js` files. I installed `flow` through _homebrew_ and followed all the steps described here https://github.com/facebook/nuclide/blob/master/pkg/nuclide/flow/README.md

**Comment by chriswise01** [comment]
This is also happening for me too. Is cmd-click supposed to work with Objective-C too or just Javascript?

**Comment by nmote** [comment]
It should work with JavaScript. However, it only works for definitions that Flow knows about, so if you have Flow errors in your code it might cause problems with click-to-symbol. Also, Flow's click-to-symbol is currently pretty slow, so you might have to be patient.

#91 might be the root cause, also.

**Comment by benjamindulau** [comment]
Doesn't work for me either.

In the console, as soon as I press "cmd" and put the mouse over something, I get the following error:

```
[2015-06-24T17:25:21.522Z] [ERROR] nuclide - Fatal error: exception End_of_file  consoleAppender.js:38 
```

Note that `flow` is in my PATH and works from the CLI (and I get the lint errors in Nuclide).

Yosemite 10.10.3
Flow installed manually from ZIP by following the official [steps](http://flowtype.org/docs/getting-started.html#installing-flow)

**Comment by scv119** [comment]
```
Fatal error: exception End_of_file
```
@nmote That's the flow's bug I mentioned.

**Comment by nmote** [comment]
Interesting, I have also observed that error in my console but click-to-symbol still works for me and I can't consistently reproduce it. I will investigate further, and thanks for reporting.

**Comment by benjamindulau** [comment]
Ok, an update on this. I installed the last version of flow => 0.12.0 and now everything is working as expected.

**Comment by nmote** [comment]
@benjamindulau: I believe Flow v0.12.0 included a fix for the `End_of_file` issue. Here are the relevant commits:

https://github.com/facebook/flow/commit/b1290009fadd8fed96778abb73404b61fe41e919

https://github.com/facebook/flow/commit/5406013aa8527de815ab0015fa4b6533f7464391

I'm going to close this, but if anybody can reproduce this issue with Flow v0.12.0, please re-open and provide repro steps.

**Comment by gunn** [comment]
It took me a while to figure out why this wasn't working for me. The reason was that I had it open as a `Babel ES6 Javascript` file using the language-babel package. How can we add support for this?

**Comment by nmote** [comment]
@gunn: if I understand correctly, in the bottom right instead of saying JavaScript it said Babel ES6 JavaScript because of the syntax highlighting provided by language-babel? And to get it working, you had to change the language to JavaScript? If so, that should be a pretty easy fix.

**Comment by gunn** [comment]
@nmote that's exactly right. There are probably quite a few syntax modes that should be supported e.g. `JavaScript (JSX)` from the react package - http://orktes.github.io/atom-react/

**Comment by nmote** [comment]
Thanks for letting us know; I will work on this.

**Comment by nmote** [comment]
@gunn: This should be fixed by 9648b729a676ae803ae2718befcd76617bec8fb4

**Comment by gunn** [comment]
@nmote confirmed! Thank you very much.
